### PR TITLE
New Module: na_ontap_ndmp

### PR DESCRIPTION
**Pull request by carchi8py**
##### SUMMARY
NetApp ONTAP NDMP services configuration

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME
- na_ontap_ndmp.py 

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/storage/netapp/na_ontap_ndmp.py [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+description:
+    - Modify NDMP Services.
+
+options:
+
+  vserver:
+    description:
+    - Name of the vserver.
+    required: true
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_ndmp.py:35]: type: str

File: lib/ansible/modules/storage/netapp/na_ontap_ndmp.py [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+    type: list
+
+  backup_log_enable:
+    description:
+    - Enable backup log.
+    type: bool
+
+  data_port_range:
+    description:
+    - Data port range.
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_ndmp.py:55]: type

File: lib/ansible/modules/storage/netapp/na_ontap_ndmp.py [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+
+  debug_enable:
+    description:
+    - Enable debug.
+    type: bool
+
+  debug_filter:
+    description:
+    - Debug filter.
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_ndmp.py:65]: type

File: lib/ansible/modules/storage/netapp/na_ontap_ndmp.py [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+
+  dump_detailed_stats:
+    description:
+    - Enable logging of VM stats for dump.
+    type: bool
+
+  dump_logical_find:
+    description:
+    - Enable logical find for dump.
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_ndmp.py:75]: type

File: lib/ansible/modules/storage/netapp/na_ontap_ndmp.py [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+    type: list
+
+  restore_vm_cache_size:
+    description:
+    - Restore VM file cache size.
+    type: int
+
+  secondary_debug_filter:
+    description:
+    - Secondary debug filter.
```
Comment (lonico) [inline lib/ansible/modules/storage/netapp/na_ontap_ndmp.py:125]: type

**Review by lonico — CHANGES_REQUESTED** [review]

**Review by lonico — APPROVED** [review]
shipit

**Comment by thedoubl3j** [comment]
@lonico @carchi8py looks like you guys have some failing tests.

File: test/units/modules/storage/netapp/test_na_ontap_ndmp.py [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    def build_ndmp_info(ndmp_details):
+        ''' build xml data for ndmp '''
+        xml = netapp_utils.zapi.NaElement('xml')
+        attributes = {
+            'num-records': 1,
+            'attributes-list': {
+                'ndmp-vserver-attributes-info': {
+                    'ignore_ctime_enabled': ndmp_details['ignore_ctime_enabled'],
+                    'backup_log_enable': ndmp_details['backup_log_enable'],
+                    'authtype': [{
```
Comment (thedoubl3j) [inline test/units/modules/storage/netapp/test_na_ontap_ndmp.py]: duplicate key in a dict here

**Review by thedoubl3j — CHANGES_REQUESTED** [review]

File: test/units/modules/storage/netapp/test_na_ontap_ndmp.py (hunk 6 above)
Comment (lonico) [reply test/units/modules/storage/netapp/test_na_ontap_ndmp.py]: This needs to be a list of 2 dictionaries.
```
[{'ndmpd-authtypes': 'plaintext'}, {'ndmpd-authtypes': 'challenge'}]
```

**Review by lonico — APPROVED** [review]
shipit

**Review by thedoubl3j — APPROVED** [review]
coverage looks good, docs look good. lgtm. shipit.

**Review by JohnLieske — APPROVED** [review]
lgtm

**Pull request merged**